tests/Doctrine/Tests/Models/OneToOneInverseSideWithAssociativeIdLoad/InverseSide.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Models\OneToOneInverseSideWithAssociativeIdLoad;
+
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\OneToOne;
+use Doctrine\ORM\Mapping\Table;
+
+/**
+ * @Entity()
+ * @Table(name="one_to_one_inverse_side_assoc_id_load_inverse")
+ */
+class InverseSide
```

Comment:
Ok so… this entity is the inverse side of the second relationship… and the owning side of the first relationship. I think your goal was to clarify things here, but I'm afraid it achieves the opposite :sweat_smile: 

Maybe you should come up with something a more realistic example, where it is obvious which entity would own which.

Reply:
This is the inverse side of the relationship the defect focuses on, which is any inverse 1-1 in an entity identified by an association. I just needed another entity for the identifier to target, so InverseSideIdTarget seemed like a reasonable name.

This was just an edit of the OneToOneInverseSideLoad model and test case, which was close to the pattern I needed. Clearly the id relationship has to be owning. The only different here is that the id column on InverseSide maps to an association and join column instead of the direct column declaration in the prior test case. If you compare the two I think it is pretty obvious what the difference is here.

The original names in the model that broke this were Action (the base type/id target here) and StateAction (the InverseSide) the other class was just State (OwningSide), which owns 1-1 Entry and Exit actions that reference StateAction. Out of context I'm not sure those names make anything any easier.

This is just a separated subtype pattern that isn't declared as such in the Doctrine ORM. It's nice to have that feature-I don't recall it being available when I first did this generated mapping in 2014 timeframe-but for the systems I'm working on this is just one of many layers generated from the same object-role model (the real breaking model produces 9k lines of Doctrine declarations sandwiched between 1400 of generated DDL and 20k lines that writes a JSON StateChange spec and reads a JSON-based query specification of the model). In other words, I very rarely interact with the Doctrine entities directly, so the final entity shape doesn't really affect me much. It doesn't matter if it is declared as a subtype or a 1-1, or if the database maps it with a separate table or combined in the same table, or even which side is owning/inverse, I just need to be able to start with any type and have the query succeed. I'm well aware this is a different perspective than using the entities as primary code artifacts, where entity aesthetics may be much more important.

Reply:
Fair enough, let's leave it as is then.